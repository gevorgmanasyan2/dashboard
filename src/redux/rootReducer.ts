import { combineReducers } from 'redux'
import * as auth from '../redux/auth/authReducer'
import * as data from './count-data/fetchDataReducer'
import * as waivers from './waiver/fetchWaiverDataReducer'
import * as location from './location/locationReducer'
import * as messages from './messages/messagesReducer'
import * as user from '../redux/my-account/userReducer'
import { authReducer } from './auth/authReducer'
import { fetchDataReducer } from './count-data/fetchDataReducer'
import { fetchWaiverDataReducer } from './waiver/fetchWaiverDataReducer'
import { locationReducer } from './location/locationReducer'
import { messagesReducer } from './messages/messagesReducer'
import { userReducer } from './my-account/userReducer'
export interface CombinedState {
  auth: auth.State
  data: data.State
  waivers: waivers.State
  location: location.State
  messages: messages.State
  user: user.State
}
const rootReducer = combineReducers({
  auth: authReducer,
  data: fetchDataReducer,
  waiver: fetchWaiverDataReducer,
  location: locationReducer,
  messages: messagesReducer,
  user: userReducer,
})

export type RootState = ReturnType<typeof rootReducer>
export default rootReducer
