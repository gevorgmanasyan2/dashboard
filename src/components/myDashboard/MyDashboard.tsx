import React, { FC } from 'react'
import DashboardActions from './dashboardActions/DashboardActions'
import './myDashboard.scss'
import Regulatory from './regulatory/Regulatory'

import { connect, useSelector } from 'react-redux'
import { RootState } from '../../redux/rootReducer'

import { UnorderedListOutlined, PlusOutlined, RightCircleOutlined } from '@ant-design/icons'

import { Link } from 'react-router-dom'

import {
  deviceUserRole,
  waiverUserRole,
  customerAdminUserRole,
  flightCountAPI,
  devicesCountAPI,
  droneCountAPI,
  userCountAPI,
} from '../../api/constants'

import { Tooltip } from 'antd'
import Title from '../title/Title'

interface MyDashboardProps {
  deviceCount: number
  droneCount: number
  flightCount: number
  userCount: number
}

const MyDashboard: FC<MyDashboardProps> = ({ deviceCount, droneCount, flightCount, userCount }) => {
  const sessionToken = useSelector((state: RootState) => state.auth.sessionToken)
  const loggedUserType = sessionToken?.roles

  const data = [
    {
      id: 1,
      icon: <RightCircleOutlined />,
      title: 'Missions',
      count: 2,
      itemTitle: 'Active',
      itemInfo: '',
      role: '',
      apiEndpoint: '',
    },
    {
      id: 2,
      icon: <RightCircleOutlined />,
      title: 'Flights',
      count: flightCount,
      itemTitle: 'This month',
      itemInfo: '',
      role: deviceUserRole,
      apiEndpoint: flightCountAPI,
    },
    {
      id: 3,
      icon: <RightCircleOutlined />,
      title: 'Casia devices',
      count: deviceCount,
      itemTitle: '',
      itemInfo: 'Software upgrade due',
      role: deviceUserRole,
      apiEndpoint: devicesCountAPI,
    },
    {
      id: 4,
      icon: <RightCircleOutlined />,
      title: 'Drones',
      count: droneCount,
      itemTitle: 'In service',
      itemInfo: '',
      role: deviceUserRole,
      apiEndpoint: droneCountAPI,
    },
    {
      id: 5,
      icon: <RightCircleOutlined />,
      title: 'Crew members',
      count: userCount,
      itemTitle: 'Active',
      itemInfo: '',
      role: customerAdminUserRole,
      apiEndpoint: userCountAPI,
    },
    {
      id: 6,
      icon: <RightCircleOutlined />,
      title: 'Locations',
      count: userCount,
      itemTitle: '',
      itemInfo: '',
      role: customerAdminUserRole,
      apiEndpoint: userCountAPI,
    },
  ]


  const checkLink = (title: string) => {
    if (title === 'Casia devices') {
      return {
        link: '/casiadevice',
        addLink: '/casiadevice',
      }
    } else if (title === 'Locations') {
      return {
        link: '/locations',
        addLink: '/addlocation',
      }
    } else {
      return {
        link: '/',
        addLink: '/casiadevice',
      }
    }
  }

  const filteredData = loggedUserType
    ? data.filter(function (el) {
      return loggedUserType.includes(el.role)
    })
    : data

  return (
    <article className="dashboard__container">
      <Title level={2} titleText="My Dashboard" />
      <div className="dashboard__container__elements">
        {filteredData.map((item) => (
          <DashboardActions
            key={item.id}
            icon={item.icon}
            title={item.title}
            count={item.count}
            itemTitle={item.itemTitle}
            itemInfo={item.itemInfo}
            apiEndpoint={item.apiEndpoint}
            loc={checkLink(item.title)}
          />
        ))}
        {loggedUserType && loggedUserType.includes(waiverUserRole) ? (
          <div className="dashboard__container__regulatory">
            <div className="dashboard__container__regulatory__element">
              <Title titleText={'Regulatory Permission'} level={3} />
              <div className="dashboard__container__regulatory__element--icon">
                <Link to={'/search-regulatory-permission'}>
                  <Tooltip title="View All">
                    <UnorderedListOutlined />
                  </Tooltip>
                </Link>
              </div>
            </div>
            <div className="dashboard__container__regulatory__components">
              <div className="dashboard__container__regulatory__components__content">
                <Regulatory />
              </div>
              <div className="dashboard__container__regulatory__icon">
                <Link to="/new-regulatory-permission">
                  <PlusOutlined />
                </Link>
              </div>
              <div className="dashboard__container__regulatory__footer">
                <Link to="/regulatory-library">
                  <p>Visit Regulatory Resource Library</p>
                </Link>
              </div>
            </div>
          </div>
        ) : (
          <div className="dashboard__container__empty" />
        )}
      </div>
    </article>
  )
}

const mapStateToProps = (state: any) => {
  return {
    deviceCount: state.data.deviceCount,
    droneCount: state.data.droneCount,
    flightCount: state.data.flightCount,
    userCount: state.data.userCount,
    waivers: state.waiver.waivers,
  }
}

export default connect(mapStateToProps)(MyDashboard)
