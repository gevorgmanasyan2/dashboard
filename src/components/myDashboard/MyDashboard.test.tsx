/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import MyDashboard from './MyDashboard'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'

it('should render MyDashboard component', () => {
  const mockItem = {
    deviceCount: 4,
    droneCount: 2,
    flightCount: 9,
    userCount: 7,
  }
  expect(
    shallow(
      <Provider store={store}>
        <MyDashboard {...mockItem} />
      </Provider>
    )
  ).toMatchSnapshot()
})
