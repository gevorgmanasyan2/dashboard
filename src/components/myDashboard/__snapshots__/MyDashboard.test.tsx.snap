// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyDashboard component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(MyDashboard)
    deviceCount={4}
    droneCount={2}
    flightCount={9}
    userCount={7}
  />
</ContextProvider>
`;
